src/cc/libclient/Monitor.cc

```diff
@@ -309,51 +307,67 @@ class Monitor::Impl: public QCRunnable {
-                    // the sum of alive clients.
-                    sumOfAllClientCounters = &sumOfAliveClientCounters;
-                }
-                else {
-                    // We have both alive and removed clients. The final sum is
-                    // equal to the sum of both.
-                    if (sumOfRemovedClientCounters.size() !=
-                            sumOfAliveClientCounters.size()) {
-                        // For each counter in sumOfAliveClientCounters,
-                        // there must be a matching counter in
-                        // fsMetrics.sumOfRemovedClientCounters.
-                        KFS_LOG_STREAM_ERROR
-                            << "Monitor: Number of counters "
-                            << "in alive and removed clients don't match!"
-                            << KFS_LOG_EOM;
+
+                if (!fsMetrics.currSumOfClientCounters.empty()) {
+                    // Calculate the difference between new and current sum of
+                    // client counters.
+                    ClientCounters diffOfCounters;
+                    if (newSumOfClientCounters.size() !=
+                            fsMetrics.currSumOfClientCounters.size()) {
+                         // For each counter in newSumOfClientCounters,
+                         // there must be a matching counter in
+                         // currSumOfClientCounters.
+                         KFS_LOG_STREAM_ERROR
+                             << "Monitor: Number of counters "
+                             << "in new and current sum of client counters"
+                                     "don't match!"
+                             << KFS_LOG_EOM;
                     }
                     for(ClientCounters::iterator countersIt =
-                            sumOfAliveClientCounters.begin();
-                            countersIt != sumOfAliveClientCounters.end();
+                            newSumOfClientCounters.begin();
+                            countersIt != newSumOfClientCounters.end();
                             ++countersIt) {
                         string counterName  = countersIt->first;
-                        Counter& counterVal = countersIt->second;
-                        counterVal += sumOfRemovedClientCounters[counterName];
+                        Counter  newCounterVal = countersIt->second;
+                        Counter  currCounterVal =
+                                fsMetrics.currSumOfClientCounters[counterName];
+                        Counter  reportVal;
+                        if(counterName.compare("Network.Sockets") == 0) {
+                            // Network.Sockets counter is not accumulative, so
+                            // report the new value.
+                            reportVal = newCounterVal;
+                        }
+                        else {
+                            reportVal = newCounterVal - currCounterVal;
```

Comment:
We decided to go with diffs in a separate discussion. Ignore this particular comment.